lib/services/local/events.go

```diff
@@ -201,9 +201,25 @@ func (e *EventsService) NewWatcher(ctx context.Context, watch types.Watch) (type
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
+
+	if err := verifyEventWatcherPrefixes(prefixes, w); err != nil {
```

Comment:
If we hit this we never close `w`. Couldn't we check if `prefixes` have overlaps before spawning the backend watcher?